libraries/checkers/usfm_checker.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import print_function, unicode_literals
+import codecs
+import os
+import traceback
+from libraries.checkers.checker import Checker
+from libraries.door43_tools.page_metrics import PageMetrics
+from libraries.usfm_tools import verifyUSFM, usfm_verses
+
+
+class UsfmChecker(Checker):
+
+    def __init__(self, preconvert_dir, converted_dir, log=None):
+        super(UsfmChecker, self).__init__(preconvert_dir, converted_dir, log=None)
+
+        # TODO blm - these are stubs until ported over to lint handler
+        self.found_books = []
+        self.commit_data = {}
+        self.rc = {}
+
+    def run(self):
+        """
+        Checks for issues with all Bibles, such as missing books or chapters
+
+        Use self.log.warning("message") to log any issues.
+        self.preconvert_dir is the directory of pre-converted files (.usfm)
+        self.converted_dir is the directory of converted files (.html)
+        :return:
+        """
+
+        lang_code = self.rc.resource.language.identifier
+        valid_lang_code = PageMetrics().validate_language_code(lang_code)
+        if not valid_lang_code:
+            self.log.warning("Invalid language code: " + lang_code)
+
+        for root, dirs, files in os.walk(self.preconvert_dir):
+            for f in files:
+                if f[-3:].lower() != 'sfm':  # only usfm files
+                    continue
+
+                file_path = os.path.join(root, f)
+                sub_path = '.' + file_path[len(self.preconvert_dir):]
+                self.parse_file(file_path, sub_path, f)
+
+        found_book_count = len(self.found_books)
+        if found_book_count == 0:
+            self.log.error("No translations found")
+        elif found_book_count == 1:
+            pass  # this is OK, presume this was a single book project
+        else:
+            self.warn_on_missing_books(self.found_books)
```

Comment:
Correct @richmahn, we will often have projects with multiple books but not all books.  This is OK.